src/core/caniuse.js

```diff
@@ -0,0 +1,229 @@
+      .map(b => b.toLowerCase())
+      .filter(isValidBrowser);
+  } else if (conf.caniuse.browsers !== "ALL") {
+    conf.caniuse.browsers = DEFAULTS.browsers;
+  }
+  Object.assign(conf.caniuse, DEFAULTS, { ...conf.caniuse });
+  function isValidBrowser(browser) {
+    if (BROWSERS.has(browser)) {
+      return true;
+    }
+    pub(
+      "warn",
+      `Ignoring invalid browser "\`${browser}\`" in [\`respecConfig.caniuse.browsers\`](https://github.com/w3c/respec/wiki/)`
+    );
+    return false;
+  }
+}
+
+/**
+ * promises to get content for canIUse table
+ * @param {Object} caniuseConf              normalized respecConfig.caniuse
+ */
+async function fetchAndCacheJson(caniuseConf) {
+  const { apiURL, feature, maxAge } = caniuseConf;
+  const url = apiURL
+    ? apiURL.replace("{FEATURE}.json", feature)
+    : `${GH_USER_CONTENT_URL}${feature}.json`;
+  // use data from cache data if valid and render
+  const cached = await cache.match(url);
+  if (cached && new Date() - cached.cacheTime < maxAge) {
+    return cached.stats;
+  }
+  // otherwise fetch new data and cache
+  const response = await window.fetch(url);
+  if (!response.ok) {
+    switch (response.status) {
+      case 404: {
+        const msg = `Couldn't find feature "${feature}" on caniuse.com? Please check the feature key on [caniuse.com](https://caniuse.com)`;
+        pub("error", msg);
+      }
+    }
+    throw new Error(
+      `Response not ok from URL ${url} (HTTP Status ${response.status})`
+    );
+  }
+  const { stats } = await response.json();
+  // set it, and forget it (there is no recovery if it throws, but that's ok).
+  cache.put(url, { stats, cacheTime: new Date() }).catch(console.error);
+  return stats;
+}
+
+/**
+ * get HTML element for the canIUse support table
+ * @param  {Object} stats     CanIUse API results
+ * @param  {Object} conf      respecConfig.caniuse
+ */
+function createTableHTML(conf, stats) {
+  let browsers = conf.browsers;
+  if (conf.browsers === "ALL") {
+    browsers = Object.keys(stats);
+  }
+
+  // render the support table
+  return hyperHTML`
+    ${browsers.map(browser =>
+      addBrowser(browser, conf.versions, stats[browser])
+    )}
+    <a href="${`http://caniuse.com/#feat=${conf.feature}`}"
+      title="Get details at caniuse.com">More info
+    </a>`;
+
+  /**
+   * add a browser and it's support to table
+   * @param {String} browser      name of browser (as in CanIUse API response)
+   * @param {Number} numVersions  number of old browser versions to show
+   * @param {Object} browserData  stats data from api response
+   */
+  function addBrowser(browser, numVersions, browserData) {
+    if (!browserData) return "";
+    const getSupport = version => {
+      const supportKeys = browserData[version]
+        .split("#", 1)[0] // don't care about footnotes.
+        .split(" ")
+        .filter(item => item);
+      let titles = [];
+      for (const key of supportKeys) {
+        if (supportTitles.has(key)) {
```

Comment:
`const tiles = supportKeys.filter(key => suppotTiles.has(key)).map(key => supportTiles.get(key) `

Reply:
even cleaner :fire: :smiley: 

Anyway, have you noticed we can't do `filter(supportTitles.has).map(supportTitles.get)` (it's a function, isn't it?). It bothers me.